### PR TITLE
Recovery service tests

### PR DESCRIPTION
**Pull request by raonismaneoto**

### CODE REVIEW

File: src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        super.tearDown();
+    }
+
+    //test case: test if the save operation works as expected by saving some objects an comparing the list returned by db.
+    @Test
+    public void testSaveOperation() throws UnexpectedException{
+        // setup
+        List<FederatedNetworkOrder> expectedOrders = new ArrayList<>();
+
+        for(int i = 0; i < 3; i++) {
```
Comment (fubica) [inline src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: Should we include another loop to include orders in all possible states?

File: src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            FederatedNetworkOrder order = testUtils.createFederatedNetwork(FEDERATED_NETWORK_ID + i, OrderState.FULFILLED);
+            expectedOrders.add(order);
+            //exercise
+            recoveryService.put(order);
+        }
+
+        //verify
+        Assert.assertEquals(expectedOrders, recoveryService.readActiveOrdersByState(OrderState.FULFILLED));
```
Comment (fubica) [inline src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: If we include orders in other states, we will need other asserts here.

File: src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    @After
+    public void tearDown() {
+        for (FederatedNetworkOrder order : orderRepository.findAll()) {
+            orderRepository.delete(order);
+        }
+        super.tearDown();
+    }
+
+    //test case: test if the save operation works as expected by saving some objects an comparing the list returned by db.
```
Comment (fubica) [inline src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: Please, correct the typo: an comparing -> and comparing

File: src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            recoveryService.put(order);
+        }
+
+        //verify
+        Assert.assertEquals(expectedOrders, recoveryService.readActiveOrdersByState(OrderState.FULFILLED));
+    }
+
+    //test case: test if the restore operation works as expected by checking if the list returned by the db contains the objects
+    // that were supposed to be there.
+    @Test
```
Comment (fubica) [inline src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: What is the difference of this test and the previous one?

Comment (raonismaneoto) [reply src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: Actually I tried to map our discussion into these two tests. Taking a better look at them, the last one doesn't make sense at all. The previous one test the two operations, save and restore, agreed? If so, i'll remove this one.

Comment (fubica) [reply src/test/java/cloud/fogbow/fns/core/datastore/orderstorage/RecoveryServiceTest.java]: Agreed.

**Pull request merged**